object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/framework/AutoValueSupport.java

```diff
@@ -0,0 +1,388 @@
+   * setters invoked. Add a CalledMethods annotation capturing this fact.
+   *
+   * @param t
+   */
+  @Override
+  public void handleToBuilder(AnnotatedExecutableType t) {
+
+    AnnotatedTypeMirror returnType = t.getReturnType();
+    ExecutableElement element = t.getElement();
+
+    String methodName = element.getSimpleName().toString();
+
+    // make sure the method is toBuilder
+    if ("toBuilder".equals(methodName)) {
+
+      TypeElement enclosingElement = (TypeElement) element.getEnclosingElement();
+      TypeMirror superclass = enclosingElement.getSuperclass();
+
+      if (FrameworkSupportUtils.hasAnnotation(enclosingElement, AutoValue.class)
+          && element.getModifiers().contains(Modifier.ABSTRACT)) {
+        handleToBuilderType(returnType, returnType.getUnderlyingType(), enclosingElement);
+      }
+
+      if (!superclass.getKind().equals(TypeKind.NONE)) {
+        TypeElement superElement = TypesUtils.getTypeElement(superclass);
+        if (FrameworkSupportUtils.hasAnnotation(superElement, AutoValue.class)) {
+          handleToBuilderType(returnType, returnType.getUnderlyingType(), superElement);
+        }
+      }
+    }
+  }
+
+  /**
+   * is element the build method for AutoValue Builder?
+   *
+   * @param element
+   * @param nextEnclosingElement
+   * @return {@code true} if return type of element is enclosing AutoValue class and method is
+   *     abstract, {@code false} otherwise
+   */
+  private boolean isBuilderBuildMethod(ExecutableElement element, Element nextEnclosingElement) {
+    return element.getModifiers().contains(Modifier.ABSTRACT)
+        && TypesUtils.getTypeElement(element.getReturnType()).equals(nextEnclosingElement);
+  }
+
+  /**
+   * Update a particular type associated with a toBuilder with the relevant CalledMethods
+   * annotation. This can be the return type of toBuilder or the corresponding generated "copy"
+   * constructor
+   *
+   * @param type type to update
+   * @param builderType type of abstract @AutoValue.Builder class
+   * @param classElement corresponding AutoValue class
+   */
+  private void handleToBuilderType(
+      AnnotatedTypeMirror type, TypeMirror builderType, Element classElement) {
+    Element builderElement = TypesUtils.getTypeElement(builderType);
+    Set<String> avBuilderSetterNames = getAutoValueBuilderSetterMethodNames(builderElement);
+    List<String> requiredProperties =
+        getAutoValueRequiredProperties(classElement, avBuilderSetterNames);
+    AnnotationMirror calledMethodsAnno =
+        createCalledMethodsForAutoValueProperties(requiredProperties, avBuilderSetterNames);
+    type.replaceAnnotation(calledMethodsAnno);
+  }
+
+  /**
+   * creates a @CalledMethods annotation for the given property names, converting the names to the
+   * corresponding setter method name in the Builder
+   *
+   * @param propertyNames
+   * @param avBuilderSetterNames
+   * @return
```

Comment:
same comment as above for `@param`, and it also applies to `@return`: if there's nothing to say, don't say anything. But better yet, do say something.